Dynatrace/dynatrace-operator
# Recreate EEC statefulset for PVC changes only

[JIRA](https://dt-rnd.atlassian.net/browse/DAQ-2893)

## Description

Hash value calculated for desired statefulset.Spec.VolumeClaimTemplates is preserved as an annotation in Annotations map.

Comparing preserved value with value calculated for new desired statefulset we can make a decision to update or recreate EEC statefulset (PVC definition was changed or not).

It's hard to compare statefulset.PVC manifests directly because for example K8S fills in PVC.VolumeMode field so Applied and then Get manifest are never equal.

## How can this be tested?
1) unittest
```
make go/test
```
2) deploy and keep updating dynakube changing EEC-PVC related properties
`useEphemeralVolume` = true/false
`tag` = 1.305.. / 1.306..
`persistentVolumeClaim` = defined / not defined 

EEC statefulset should be recreated only if `useEphemeralVolume` or `persistentVolumeClaim` is changed.
```
  templates:
    extensionExecutionController:
      #useEphemeralVolume: true
      useEphemeralVolume: false
      imageRef:
        repository: ....
        #tag: 1.306.0.20241128-144422
        tag: 1.305.0.20241029-011002
      #persistentVolumeClaim:
      #  accessModes:
      #  - ReadWriteOnce
      #  storageClassName: "standard"
      #  resources:
      #    requests:
      #      storage: 2G 
```
